In [1]:
import tensorflow as tf 

In [2]:
tf

<module 'tensorflow' from 'C:\\Users\\ASUS\\anaconda3\\Lib\\site-packages\\tensorflow\\__init__.py'>

In [3]:

from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Flatten
from tensorflow.keras.datasets import mnist
from tensorflow.keras.utils import to_categorical


In [4]:
(x_train,y_train) , (x_test,y_test) = mnist.load_data()

In [5]:
# convert input(255 pixel image) into 1 pixel 
x_train = x_train/255.0
x_test = x_test/255.0

In [6]:
#convert output into onehot by categorical(it will contain 10 labels)
y_train = to_categorical(y_train , 10)
y_test = to_categorical(y_test , 10)

In [7]:
model = Sequential([
    Flatten(input_shape = (28,28)) , 
    Dense(128 , activation = "relu") , 
    Dense(10 , activation = "softmax")
])

C:\Users\ASUS\anaconda3\Lib\site-packages\keras\src\layers\reshaping\flatten.py:37: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(**kwargs)


In [8]:
model.compile(
    optimizer="adam",
    loss="categorical_crossentropy",
    metrics=['accuracy']
)

In [9]:
model.fit(x_train , y_train , epochs = 5 ,batch_size = 32)

Epoch 1/5
1875/1875 ━━━━━━━━━━━━━━━━━━━━ 7s 3ms/step - accuracy: 0.9255 - loss: 0.2601
Epoch 2/5
1875/1875 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - accuracy: 0.9657 - loss: 0.1166
Epoch 3/5
1875/1875 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - accuracy: 0.9759 - loss: 0.0797
Epoch 4/5
1875/1875 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - accuracy: 0.9815 - loss: 0.0600
Epoch 5/5
1875/1875 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - accuracy: 0.9859 - loss: 0.0463


In [10]:
test_loss , test_accuracy = model.evaluate(x_test , y_test)

313/313 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.9773 - loss: 0.0747


In [11]:
print(f"Accuracy = {test_accuracy}")

Accuracy = 0.9772999882698059


CONCLUSION 


-We have trained the model on mnist dataset inorder to predict/detect the Handwritten letters or Handwritting 


-The model gives 97% Accuracy when tested on the Testing data 